src/lib/mina/fetch.ts

```diff
@@ -1049,7 +1128,11 @@ async function makeGraphqlRequest<TDataResponse = any>(
     try {
       let response = await fetch(url, {
         method: 'POST',
-        headers: { 'Content-Type': 'application/json' },
+        headers: {
+          'Content-Type': 'application/json',
+          ...(isArchiveUrl(url) ? archiveDefaultHeaders : minaDefaultHeaders),
```

Comment:
I think it's cleaner to pass this function the exact headers we want to use.  That way we can get rid of the `isArchiveUrl` logic.

We know that `fetchTransactionStatus` goes to a Mina node and `fetchEvents` goes to an archive endpoint.  So we can set the correct headers directly in those methods:


```ts
// fetchEvents, for example
let [response, error] = await makeGraphqlRequest<EventQueryResponse>(
    getEventsQuery(
      publicKey,
      tokenId ?? TokenId.toBase58(TokenId.default),
      filterOptions
    ),
    graphqlEndpoint,
    networkConfig.archiveFallbackEndpoints,
    { ...archiveDefaultHeaders, ...headers}
```